File: r-package/R/codigos.R
#' Códigos e nomes dos estados brasileiros (2022)
#'
#' Tabela com a relação entre os códigos e nomes dos estados brasileiros. Os
#' códigos foram convertidos para caracteres; os nomes foram convertidos para
#' caracteres ASCII em caixa alta.
#'
#' @format
#' Um data frame com 27 linhas e 3 colunas:
#' - `codigo_estado` - código do estado;
#' - `nome_estado` - nome do estado;
#' - `abrev_estado` - abreviação do nome do estado.
#'
#' @source <https://www.ibge.gov.br/explica/codigos-dos-municipios.php>
#'
#' @seealso [codigos_municipios]
#'
#' @name codigos_estados
#' @export
"codigos_estados"

#' Códigos e nomes dos municípios brasileiros (2022)
#'
#' Tabela com a relação entre os códigos e nomes dos municípios brasileiros. Os
#' códigos foram convertidos para caracteres; os nomes foram convertidos para
#' caracteres ASCII em caixa alta.
#'
#' @format
#' Um data frame com 5570 linhas e 3 colunas:
#' - `codigo_estado` - código do estado em que o município está localizado;
#' - `codigo_muni` - código do município;
#' - `nome_muni` - nome do município.
#'
#' @source <https://www.ibge.gov.br/explica/codigos-dos-municipios.php>
#'
#' @seealso [codigos_estados]
#'
#' @name codigos_municipios
#' @export
"codigos_municipios"

File: python-package/src/enderecobr/codigos.py
# src/enderecobr/codigos.py

"""
Códigos e nomes dos estados e municípios brasileiros (2022)

Fonte:
  https://www.ibge.gov.br/explica/codigos-dos-municipios.php

Formato:
- Estados:
    - codigo_estado: código do estado (str)
    - nome_estado: nome do estado (ASCII maiúsculo)
    - abrev_estado: abreviação do nome do estado (str)

- Municípios:
    - codigo_estado: código do estado (str)
    - codigo_muni: código do município (str)
    - nome_muni: nome do município (ASCII maiúsculo)
"""

import pandas as pd
import importlib.resources as resources

# Estados
with resources.as_file(resources.files("enderecobr") / "data" / "codigos_brasil.xls") as xls_path:
    codigos_estados = pd.read_excel(
        xls_path,
        sheet_name="estados",
        dtype=str,
        engine="openpyxl"
    )

# Municípios
with resources.as_file(resources.files("enderecobr") / "data" / "codigos_brasil.xls") as xls_path:
    codigos_municipios = pd.read_excel(
        xls_path,
        sheet_name="municipios",
        dtype=str,
        engine="openpyxl"
    )


def carregar_codigos_estados():
    return codigos_estados.copy()


def carregar_codigos_municipios():
    return codigos_municipios.copy()
